UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs

```diff
@@ -306,7 +306,7 @@ public virtual void AddBodyPartSlot(ItemSlot ItemSlot)
 		public virtual void RemoveFromBodyThis()
 		{
 			BodyPartRemovalChecks();
-			RootBodyPartContainer parent = this.GetParent();
+			dynamic parent = this.GetParent();
```

Comment:
what's ``dynamic``?

Reply:
it's basically python flexibility

Reply:
why does it need to be type flexible?

Reply:
the problem is, root body part container and body part both could contain body parts, so it could Either or

Reply:
aaah I see. So this GetParent could either return a RootBodyPartContainer or BodyPart. This is screaming bad implementation to me, why aren't those clases parented under the same abstract class so we make this depend on abstracts instead  of concretes than can be of different type?

Reply:
https://en.wikipedia.org/wiki/Dependency_inversion_principle

Reply:
idk, just wanted to try dynamic it works pretty well

Reply:
also say that having to write a bunch of code, because only came across the issue once it was all mostly written up

Reply:
*also saved having to write a bunch of code

Reply:
yes, I understand and I guess it is good enough for now. Can you slap a TODO comment there so we can revisit it later? I frankly don't feel comfortable with it being like that.